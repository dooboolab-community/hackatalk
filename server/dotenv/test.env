JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
