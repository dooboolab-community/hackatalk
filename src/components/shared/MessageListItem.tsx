import React, { SFC } from 'react';
import { TouchableOpacity, View } from 'react-native';

import { Ionicons } from '@expo/vector-icons';
import { MessageProps } from '../../types';
import moment from 'moment';
import styled from 'styled-components/native';
import { useThemeContext } from '@dooboo-ui/native-theme';

const WrapperPeer = styled.View<{ isSame: boolean }>`
  min-height: 48px;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  margin-left: 20px;
  margin-right: 8px;
  margin-top: ${({ isSame }): number => (isSame ? 8 : 20)}px;
`;

const StyledImageSender = styled.Image`
  width: 32px;
  height: 32px;
`;

const StyledTextPeerMessageContainer = styled.View`
  margin-right: 8px;
  background-color: ${({ theme }): string => theme.peerMessageBackground};
  padding: 12px;
  width: 100%;
`;

const StyledPeerTextMessage = styled.Text`
  font-size: 14px;
  color: ${({ theme }): string => theme.peerMessageText};
`;

const StyledTextPeerName = styled.Text`
  font-size: 12px;
  color: ${({ theme }): string => theme.fontColor};
  margin-bottom: 2px;
`;

const StyledTextPeerDate = styled.Text`
  font-size: 12px;
  color: ${({ theme }): string => theme.fontColor};
  margin-right: 20px;
  margin-top: 4px;
`;

const WrapperMy = styled.View`
  min-height: 48px;
  width: 100%;
  margin-top: 20px;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-end;
`;

const StyledTextDate = styled.Text`
  font-size: 12px;
  color: ${({ theme }): string => theme.status};
  margin-top: 4px;
  margin-right: 20px;
`;

const StyledMyTextMessage = styled.Text`
  font-size: 14px;
  color: ${({ theme }): string => theme.myMessageText};
`;

const StyledMyMessage = styled.View`
  background-color: ${({ theme }): string => theme.myMessageBackground};
  margin-right: 20px;
  margin-left: 28px;
  padding: 12px;
  border-radius: 3px;
`;

interface Props {
  item: MessageProps;
  prevItem?: MessageProps;
  onPressPeerImage?: () => void;
  testID?: string;
}

interface ImageSenderProps {
  photoURL?: string;
  isSamePeerMsg: boolean;
  fontColor: string;
}

const myFakeUid = '2'; // TODO: temporary

const ImageSenderComp: SFC<ImageSenderProps> = ({
  photoURL,
  isSamePeerMsg,
  fontColor,
}) => {
  if (photoURL !== '') {
    return <StyledImageSender source={{ uri: photoURL }} />;
  } else if (isSamePeerMsg) {
    return <View style={{ width: 40 }} />;
  }
  return (
    <View style={{ width: 40 }}>
      <Ionicons name="ios-person" size={40} color={fontColor} />
    </View>
  );
};

function MessageListItem(props: Props): React.ReactElement {
  const { theme } = useThemeContext();
  const {
    item: {
      sender: { uid, displayName, photoURL },
      messageType,
      // @ts-ignore
      message,
      created,
    },
    prevItem,
    onPressPeerImage,
    testID,
  } = props;
  const isSamePeerMsg = prevItem && prevItem.sender.uid === uid;
  if (uid !== myFakeUid) {
    return (
      <WrapperPeer isSame={!!isSamePeerMsg}>
        <View style={{ marginRight: 8, width: 40 }}>
          <TouchableOpacity testID={testID} onPress={onPressPeerImage}>
            <ImageSenderComp
              photoURL={photoURL}
              isSamePeerMsg={!!isSamePeerMsg}
              fontColor={theme.fontColor}
            />
          </TouchableOpacity>
        </View>
        <View style={{ flexDirection: 'column', maxWidth: '90%' }}>
          {isSamePeerMsg ? (
            <View />
          ) : (
            <StyledTextPeerName>{displayName}</StyledTextPeerName>
          )}
          <StyledTextPeerMessageContainer>
            <StyledPeerTextMessage>{message}</StyledPeerTextMessage>
          </StyledTextPeerMessageContainer>
          <StyledTextPeerDate>
            {created
              ? `${moment(created).hour()} : ${moment(created).minutes()}`
              : '0 : 0'}
          </StyledTextPeerDate>
        </View>
      </WrapperPeer>
    );
  }
  return (
    <WrapperMy>
      <StyledMyMessage>
        <StyledMyTextMessage>{message}</StyledMyTextMessage>
      </StyledMyMessage>
      <StyledTextDate>
        {created
          ? `${moment(created).hour()} : ${moment(created).minutes()}`
          : '0 : 0'}
      </StyledTextDate>
    </WrapperMy>
  );
}

MessageListItem.defaultProps = {
  item: {
    id: '',
    sender: {
      uid: '0',
      displayName: 'sender111',
      thumbURL: '',
      photoURL: '',
      statusMsg: '',
    },
    message: 'hello1',
    created: new Date('1 1 2019'),
    updated: new Date('1 1 2019'),
  },
};

export default MessageListItem;
