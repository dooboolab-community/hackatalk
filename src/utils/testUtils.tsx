import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
