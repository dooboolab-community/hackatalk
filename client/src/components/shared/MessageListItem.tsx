import React, { SFC } from 'react';
import { TouchableOpacity, View } from 'react-native';

import { IC_NO_IMAGE } from '../../utils/Icons';
import { MessageProps } from '../../types';
import moment from 'moment';
import styled from 'styled-components/native';
import { useThemeContext } from '@dooboo-ui/theme';

const WrapperPeer = styled.View<{ isSame: boolean }>`
  min-height: 48px;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  margin-left: 20px;
  margin-right: 8px;
  margin-top: ${({ isSame }): number => (isSame ? 8 : 20)}px;
`;

const StyledImage = styled.Image`
  width: 32px;
  height: 32px;
  border-radius: 16px;
`;

const StyledImageSender = styled.Image`
  width: 32px;
  height: 32px;
`;

const StyledTextPeerMessageContainer = styled.View`
  margin-right: 8px;
  background-color: ${({ theme }): string => theme.peerMessageBackground};
  padding: 12px;
  width: 100%;
`;

const StyledPeerTextMessage = styled.Text`
  font-size: 14px;
  color: ${({ theme }): string => theme.peerMessageText};
`;

const StyledTextPeerName = styled.Text`
  font-size: 12px;
  color: ${({ theme }): string => theme.fontColor};
  margin-bottom: 2px;
`;

const StyledTextPeerDate = styled.Text`
  font-size: 12px;
  color: ${({ theme }): string => theme.fontColor};
  margin-right: 20px;
  margin-top: 4px;
`;

const WrapperMy = styled.View`
  min-height: 48px;
  width: 100%;
  margin-top: 20px;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-end;
`;

const StyledTextDate = styled.Text`
  font-size: 12px;
  color: ${({ theme }): string => theme.fontColor};
  margin-top: 4px;
  margin-right: 20px;
`;

const StyledMyTextMessage = styled.Text`
  font-size: 14px;
  color: ${({ theme }): string => theme.myMessageText};
`;

const StyledMyMessage = styled.View`
  background-color: ${({ theme }): string => theme.myMessageBackground};
  margin-right: 20px;
  margin-left: 28px;
  padding: 12px;
  border-radius: 3px;
`;

interface Props {
  item: MessageProps;
  prevItem?: MessageProps;
  nextItem?: MessageProps;
  onPressPeerImage?: () => void;
  testID?: string;
}

interface ImageSenderProps {
  thumbURL?: string | null;
  isSamePeerMsg: boolean;
  fontColor: string;
}

const myFakeId = '2'; // TODO: temporary

function shouldShowDate(
  currentDate: string | undefined,
  nextDate: string | undefined,
): boolean {
  if (!currentDate || !nextDate) return false;

  const currentMoment = moment(currentDate);
  const nextMoment = moment(nextDate);

  const diffNextSeconds = nextMoment.diff(currentMoment, 'seconds');

  return diffNextSeconds < 60 && (nextMoment.minute === currentMoment.minute);
}

const ImageSenderComp: SFC<ImageSenderProps> = ({
  thumbURL,
  isSamePeerMsg,
  fontColor,
}) => {
  if (isSamePeerMsg) {
    return <View style={{ width: 40 }} />;
  } else if (thumbURL) {
    return <StyledImageSender source={{ uri: thumbURL }} />;
  }

  return (
    <View
      style={{
        width: 40,
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
      }}
    >
      <StyledImage
        source={IC_NO_IMAGE}
      />
    </View>
  );
};

function MessageListItem(props: Props): React.ReactElement {
  const { theme } = useThemeContext();

  const {
    item: {
      sender: { id: senderId, nickname, photoURL: thumbURL },
      messageType,
      // @ts-ignore
      message,
      created,
    },
    prevItem,
    nextItem,
    onPressPeerImage,
    testID,
  } = props;

  const isSamePeerMsg = prevItem && prevItem.sender.id === senderId;
  const showDate = shouldShowDate(created, nextItem?.created);

  if (senderId !== myFakeId) {
    return (
      <WrapperPeer isSame={!!isSamePeerMsg}>
        <View style={{ marginRight: 8, width: 40 }}>
          <TouchableOpacity testID={testID} onPress={onPressPeerImage}>
            <ImageSenderComp
              thumbURL={thumbURL}
              isSamePeerMsg={!!isSamePeerMsg}
              fontColor={theme.fontColor}
            />
          </TouchableOpacity>
        </View>
        <View style={{ flexDirection: 'column', maxWidth: '90%' }}>
          {isSamePeerMsg ? (
            <View />
          ) : (
            <StyledTextPeerName>{nickname}</StyledTextPeerName>
          )}
          <StyledTextPeerMessageContainer>
            <StyledPeerTextMessage>{message}</StyledPeerTextMessage>
          </StyledTextPeerMessageContainer>
          {
            !showDate
              ? <StyledTextPeerDate>
                {created
                  ? `${moment(created).hour()} : ${moment(created).minutes()}`
                  : '0 : 0'}
              </StyledTextPeerDate>
              : null
          }
        </View>
      </WrapperPeer>
    );
  }

  return (
    <WrapperMy>
      <StyledMyMessage>
        <StyledMyTextMessage>{message}</StyledMyTextMessage>
      </StyledMyMessage>
      <StyledTextDate>
        {created
          ? `${moment(created).hour()} : ${moment(created).minutes()}`
          : '0 : 0'}
      </StyledTextDate>
    </WrapperMy>
  );
}

MessageListItem.defaultProps = {
  item: {
    id: '',
    sender: {
      id: '0',
      nickname: 'sender111',
      thumbURL: '',
      photoURL: '',
      statusMessage: '',
    },
    message: 'hello1',
    created: '2020-01-01 12:00',
    updated: '2020-01-01 12:00',
  },
};

export default MessageListItem;
