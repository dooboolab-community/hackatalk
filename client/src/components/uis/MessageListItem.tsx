import React, {FC} from 'react';
import {TouchableOpacity, View} from 'react-native';

import {IC_NO_IMAGE} from '../../utils/Icons';
import Image from 'react-native-scalable-image';
import {Message} from '../../types/graphql';
import {getString} from '../../../STRINGS';
import moment from 'moment';
import styled from 'styled-components/native';
import {useTheme} from 'dooboo-ui';

const WrapperPeer = styled.View<{isSame: boolean}>`
  min-height: 48px;
  flex-direction: row;
  align-items: flex-start;
  justify-content: flex-start;
  margin-left: 20px;
  margin-right: 8px;
  margin-top: ${({isSame}): number => (isSame ? 8 : 20)}px;
  width: 100%;
`;

const StyledImage = styled.Image`
  width: 32px;
  height: 32px;
  border-radius: 16px;
`;

const StyledImageSender = styled.Image`
  width: 32px;
  height: 32px;
  border-radius: 16px;
`;

const StyledTextPeerMessageContainer = styled.View`
  margin-right: 8px;
  background-color: ${({theme}) => theme.peerMessageBackground};
  padding: 12px;
`;

const StyledPeerTextMessage = styled.Text`
  font-size: 14px;
  color: ${({theme}) => theme.peerMessageText};
`;

const StyledPhotoContainer = styled.View`
  border-color: ${({theme}) => theme.disabled};
  border-width: 1px;
`;

const StyledTextPeerName = styled.Text`
  font-size: 12px;
  color: ${({theme}) => theme.text};
  margin-bottom: 2px;
`;

const StyledTextPeerDate = styled.Text`
  font-size: 12px;
  color: ${({theme}) => theme.text};
  margin-right: 20px;
  margin-top: 4px;
`;

const WrapperMy = styled.View`
  min-height: 48px;
  width: 100%;
  margin-top: 20px;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-end;
`;

const StyledTextDate = styled.Text`
  font-size: 12px;
  color: ${({theme}) => theme.text};
  margin-top: 4px;
  margin-right: 20px;
`;

const StyledMyTextMessage = styled.Text`
  font-size: 14px;
  color: ${({theme}) => theme.myMessageText};
`;

const StyledMyMessage = styled.View`
  background-color: ${({theme}) => theme.myMessageBackground};
  margin-right: 20px;
  margin-left: 28px;
  padding: 12px;
  border-radius: 3px;
`;

interface Props<T> {
  userId?: string;
  item: T;
  prevItem?: T;
  nextItem?: T;
  onPressPeerImage?: () => void;
  onPressMessageImage?: (index: number) => void;
  testID?: string;
}

interface ImageSenderProps {
  thumbURL?: string | null;
  isSamePeerMsg: boolean;
  fontColor: string;
}

function shouldShowDate(
  currentDate: string | undefined,
  nextDate: string | undefined,
): boolean {
  if (!currentDate || !nextDate) return false;

  const currentMoment = moment(currentDate);
  const nextMoment = moment(nextDate);

  const diffNextSeconds = nextMoment.diff(currentMoment, 'seconds');

  return diffNextSeconds < 60 && nextMoment.minute === currentMoment.minute;
}

const ImageSender: FC<ImageSenderProps> = ({thumbURL, isSamePeerMsg}) => {
  if (isSamePeerMsg) return <View style={{width: 40}} />;
  else if (thumbURL) return <StyledImageSender source={{uri: thumbURL}} />;

  return (
    <View
      style={{
        width: 40,
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
      }}>
      <StyledImage source={IC_NO_IMAGE} />
    </View>
  );
};

function MessageListItem<T>(props: Props<T & Message>): React.ReactElement {
  const {theme} = useTheme();

  const {
    item: {id, sender, text, createdAt, imageUrls},
    prevItem,
    nextItem,
    onPressPeerImage,
    onPressMessageImage,
    testID,
    userId,
  } = props;

  const isSamePeerMsg = prevItem?.sender?.id === sender?.id;
  const showDate = shouldShowDate(createdAt, nextItem?.createdAt);

  if (sender?.id !== userId)
    return (
      <WrapperPeer isSame={!!isSamePeerMsg}>
        <View style={{marginRight: 8, width: 40}}>
          <TouchableOpacity testID={testID} onPress={onPressPeerImage}>
            <ImageSender
              thumbURL={sender?.thumbURL}
              isSamePeerMsg={!!isSamePeerMsg}
              fontColor={theme.text}
            />
          </TouchableOpacity>
        </View>
        <View style={{flexDirection: 'column', maxWidth: '80%'}}>
          {isSamePeerMsg ? (
            <View />
          ) : (
            <StyledTextPeerName>
              {sender?.nickname || sender?.name || getString('NO_NAME')}
            </StyledTextPeerName>
          )}
          <View
            style={{
              flexDirection: 'row',
              justifyContent: 'flex-start',
            }}>
            <StyledTextPeerMessageContainer>
              {imageUrls && imageUrls.length > 0 ? (
                <StyledPhotoContainer>
                  <TouchableOpacity
                    onPress={() =>
                      onPressMessageImage && onPressMessageImage(0)
                    }>
                    <Image
                      key={id}
                      width={240}
                      source={{uri: `${imageUrls[0]}?id=${id}`}}
                    />
                  </TouchableOpacity>
                </StyledPhotoContainer>
              ) : (
                <StyledPeerTextMessage>{text}</StyledPeerTextMessage>
              )}
            </StyledTextPeerMessageContainer>
          </View>
          {!showDate ? (
            <StyledTextPeerDate>
              {createdAt ? `${moment(createdAt).fromNow()}` : '0 : 0'}
            </StyledTextPeerDate>
          ) : null}
        </View>
      </WrapperPeer>
    );

  return (
    <WrapperMy>
      <StyledMyMessage>
        {imageUrls && imageUrls.length > 0 ? (
          <StyledPhotoContainer>
            <TouchableOpacity
              onPress={() => onPressMessageImage && onPressMessageImage(0)}>
              <Image
                key={id}
                width={240}
                source={{uri: `${imageUrls[0]}?id=${id}`}}
              />
            </TouchableOpacity>
          </StyledPhotoContainer>
        ) : (
          <StyledMyTextMessage>{text}</StyledMyTextMessage>
        )}
      </StyledMyMessage>
      <StyledTextDate>
        {createdAt ? `${moment(createdAt).fromNow()}` : '0 : 0'}
      </StyledTextDate>
    </WrapperMy>
  );
}

export default MessageListItem;
