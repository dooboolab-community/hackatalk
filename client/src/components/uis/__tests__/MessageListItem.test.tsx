import {MockPayloadGenerator, createMockEnvironment} from 'relay-test-utils';
import React, {Suspense} from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {graphql, useLazyLoadQuery} from 'react-relay';

import MessageListItem from '../MessageListItem';
import {MessageListItemTestQuery} from '../../../__generated__/MessageListItemTestQuery.graphql';
import {ProfileModal_user$key} from '../../../__generated__/ProfileModal_user.graphql';
import {Text} from 'react-native';
import {createTestElement} from '../../../../test/testUtils';

const SAMPLE_MESSAGE = {
  messageType: 'text',
  text: 'Hello there!',
  createdAt: '2021-03-19T01:11:48.518Z',
  sender: {
    id: 'test-user-111',
    name: 'Alice',
    nickname: 'alice',
    photoURL: 'https://example.com/large.jpg',
    thumbURL: 'https://example.com/profile.jpg',
    hasBlocked: false,
    statusMessage: 'Hello WOrld!',
  },
};

type QueryWrapperProps = {
  onPressPeerImage?: (sender: ProfileModal_user$key) => void;
  onPressMessageImage?: (index: number) => void;
};

const QueryWrapper: React.FC<QueryWrapperProps> = (props) => {
  const {myData} = useLazyLoadQuery<MessageListItemTestQuery>(
    graphql`
      query MessageListItemTestQuery {
        myData: message(id: "test-id") {
          ...MessageListItem_message
        }
      }
    `,
    {},
  );

  if (!myData) return <Text>FAILED TO FETCH</Text>;

  return (
    <Suspense fallback={<Text>TEST FALLBACK</Text>}>
      <MessageListItem
        {...props}
        testID="chat-list-item0"
        item={myData}
        prevItemSenderId="previous-message-id"
        nextItemDate="2021-03-19T00:57:16.762Z"
      />
    </Suspense>
  );
};

const mockEnvironment = createMockEnvironment();

mockEnvironment.mock.queueOperationResolver((operation) =>
  MockPayloadGenerator.generate(operation, {
    Message(_, generateId) {
      return {
        ...SAMPLE_MESSAGE,
        id: `test-message-${generateId()}`,
      };
    },
  }),
);

describe('[MessageListItem] rendering test', () => {
  it('renders [peerMessage] as expected', async () => {
    const component = createTestElement(<QueryWrapper />, {
      environment: mockEnvironment,
    });

    const screen = render(component);

    // Wait until query is resolved.
    await waitFor(() => {
      expect(screen.queryByText('TEST FALLBACK')).toBeNull();
    });

    const json = screen.toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });

  it('renders [peerMessage] with URL as expected', async () => {
    const component = createTestElement(<QueryWrapper />, {
      environment: mockEnvironment,
    });

    const screen = render(component);

    const senderImage = await screen.findByHintText('sender image');

    expect(senderImage.props.source.uri).toBe(
      'https://example.com/profile.jpg',
    );
  });
});

describe('[MessageListItem] interaction', () => {
  it('should fireEvent when peer image is pressed', async () => {
    const mockPressPeerImage = jest.fn();

    const component = createTestElement(
      <QueryWrapper onPressPeerImage={mockPressPeerImage} />,
      {
        environment: mockEnvironment,
      },
    );

    const screen = render(component);
    const touchPeerImage = screen.getByTestId('chat-list-item0');

    fireEvent.press(touchPeerImage);
    expect(mockPressPeerImage).toHaveBeenCalledTimes(1);
  });
});
