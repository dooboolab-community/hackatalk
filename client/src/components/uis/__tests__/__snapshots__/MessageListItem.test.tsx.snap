// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MessageListItem] rendering test renders [peerMessage] as expected 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    shouldShowDatePeer={true}
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginLeft": 20,
          "marginRight": 8,
          "marginTop": 8,
          "minHeight": 48,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "marginRight": 8,
          "width": 40,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="chat-list-item0"
      >
        <Image
          accessibilityHint="sender image"
          accessible={true}
          source={
            Object {
              "uri": "https://example.com/profile.jpg",
            }
          }
          style={
            Array [
              Object {
                "borderBottomLeftRadius": 16,
                "borderBottomRightRadius": 16,
                "borderTopLeftRadius": 16,
                "borderTopRightRadius": 16,
                "height": 32,
                "width": 32,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "maxWidth": "80%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 12,
              "marginBottom": 2,
            },
            undefined,
          ]
        }
      >
        alice
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#E2F2FF",
                "marginRight": 8,
                "paddingBottom": 12,
                "paddingLeft": 12,
                "paddingRight": 12,
                "paddingTop": 12,
              },
              undefined,
            ]
          }
        >
          <Text
            selectable={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 14,
                },
                undefined,
              ]
            }
          >
            <Text>
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                Hello there!
              </Text>
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
